tests/Oryx.BuildImage.Tests/OryxCommandTest.cs

```diff
@@ -3,21 +3,76 @@
 // Licensed under the MIT license.
 // --------------------------------------------------------------------------------------------
 
-using Xunit;
-using Xunit.Abstractions;
+using System;
+using Microsoft.Oryx.BuildScriptGenerator.DotNetCore;
 using Microsoft.Oryx.BuildScriptGenerator.Node;
 using Microsoft.Oryx.BuildScriptGenerator.Php;
 using Microsoft.Oryx.Common;
 using Microsoft.Oryx.Tests.Common;
-using Microsoft.Oryx.BuildScriptGenerator.DotNetCore;
-using System;
+using Xunit;
+using Xunit.Abstractions;
 
 namespace Microsoft.Oryx.BuildImage.Tests
 {
     public class OryxCommandTest : SampleAppsTestBase
     {
+        private const int UnsupportedPlatform = 2;
+        private const int UnsupportedPlatformVersion = 3;
+
         public OryxCommandTest(ITestOutputHelper output) : base(output) { }
 
+        [Fact]
+        public void Build_ReturnsExpectedErrorCode_ForUnsupportedPlatformException()
+        {
+            // Arrange
+            var script = new ShellScriptBuilder()
+               .AddCommand("mkdir /tmp/app")
+               .AddCommand("oryx build /tmp/app -o /tmp/out")
+               .ToString();
+
+            // Act
+            var result = _dockerCli.Run(new DockerRunArguments
+            {
+                ImageId = Settings.BuildImageName,
+                CommandToExecuteOnRun = "/bin/bash",
+                CommandArguments = new[] { "-c", script }
+            });
+
+            // Assert
+            RunAsserts(
+                () =>
+                {
+                    Assert.Equal(UnsupportedPlatform, result.ExitCode);
+                },
+                result.GetDebugInfo());
+        }
+
+        [Fact]
+        public void Build_ReturnsExpectedErrorCode_ForUnsupportedVersionException()
```

Comment:
nit: its trivial, but it would be nice to have unsupported versions from other platfo0rms as well